### DISCUSSION TITLE
Composer version support

### DISCUSSION THREAD

**Discussion by pimjansen** [renovatebot/renovate#19179] [Request Help]
Not sure if there is a solution for this but here is my problem. To summarize this i have a dependency from `Laminas` which is locked on a composer API version < 2.3.0 which means it only supports the `Composer LTS` version 2.2.

The current Renovate docker image runs a composer version newer as this (latest i assume) on which i cant run it for this project. Is there a way to specify the composer version or to bypass this?

Run output:

```
Command failed: composer update phpro/grumphp --with-dependencies --ignore-platform-req='ext-*' --ignore-platform-req='lib-*' --no-ansi --no-interaction --no-scripts --no-autoloader --no-plugins
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laminas/laminas-dependency-plugin is locked to version 2.5.0 and an update of this package was not requested.
    - laminas/laminas-dependency-plugin 2.5.0 requires composer-plugin-api >=1.1.0 <2.3.0 -> found composer-plugin-api[2.3.0] but it does not match the constraint.
  Problem 2
    - laminas/laminas-dependency-plugin 2.5.0 requires composer-plugin-api >=1.1.0 <2.3.0 -> found composer-plugin-api[2.3.0] but it does not match the constraint.
    - rwoverdijk/assetmanager 3.0.0 requires laminas/laminas-dependency-plugin ^2.0 -> satisfiable by laminas/laminas-dependency-plugin[2.5.0].
    - rwoverdijk/assetmanager is locked to version 3.0.0 and an update of this package was not requested.
 ```

**Comment by viceice** [comment]
use constraints

**Comment by pimjansen** [comment]
> use constraints 

Constrains where? The docker image of renovate has a fixed composer version right?

**Reply by rarkins** [reply]
Look up the docs for binarySource=install, this should be the default value for Renovate's images now

**Reply by pimjansen** [reply]
@rarkins thanks for the hint but how can i specify the version? This since i have the same "problem" with the PHP version which is globally bumped to 8.2 where i have some legacy apps still running 7.4.

**Reply by rarkins** [reply]
Use `constraints`: https://docs.renovatebot.com/configuration-options/#constraints
